revolution/drivers.py

```diff
@@ -209,3 +211,334 @@ def voltages(self) -> dict[InputChannel, float]:
+        "BUVL":             {"addr":0x0F, "size":2, "permission":"RW", "signed": False, "data":0x0000},
+        "TEMP_LIMIT":       {"addr":0x10, "size":2, "permission":"RW", "signed": True, "data":0x7FFF},
+        "PWR_LIMIT":        {"addr":0x11, "size":2, "permission":"RW", "signed": False, "data":0xFFFF},
+        "MANUFACTURER_ID":  {"addr":0x3E, "size":2, "permission":"RO", "signed": False, "data":0x5449},
+        "DEVICE_ID":        {"addr":0x3F, "size":2, "permission":"RO", "signed": False, "data":0x2291},
+    }
+    _adc_averaging_count_to_bin = bidict({1:0, 4:1, 16:2, 64:3, 128:4, 256:5, 512:6, 1024:7})
+    _conversion_time_us_to_bin = bidict({50:0, 84:1, 150:2, 280:3, 540:4, 1052:5, 2074:6, 4120:7})
+    _mode_str_to_bin = bidict({
+        "SHUTDOWN":             0x0, #Shutdown
+        "TRG_VOLT_ONLY":        0x1, #Triggered bus voltage, single shot
+        "TRG_CURR_ONLY":        0x2, #Triggered shunt voltage, single shot
+        "TRG_VOLT_CURR":        0x3, #Triggered shunt voltage and bus voltage, single shot
+        "TRG_TEMP_ONLY":        0x4, #Triggered temperature, single shot
+        "TRG_VOLT_TEMP":        0x5, #Triggered temperature and bus voltage, single shot
+        "TRG_CURR_TEMP":        0x6, #Triggered temperature and shunt voltage, single shot
+        "TRG_VOLT_CURR_TEMP":   0x7, #Triggered bus voltage, shunt voltage and temperature, single shot
+        "SHUTDOWN":             0x8, #Shutdown
+        "CONT_VOLT_ONLY":       0x9, #Continuous bus voltage only
+        "CONT_CURR_ONLY":       0xA, #Continuous shunt voltage only
+        "CONT_VOLT_CURR":       0xB, #Continuous shunt and bus voltage
+        "CONT_TEMP_ONLY":       0xC, #Continuous temperature only
+        "CONT_VOLT_TEMP":       0xD, #Continuous bus voltage and temperature
+        "CONT_CURR_TEMP":       0xE, #Continuous temperature and shunt voltage
+        "CONT_VOLT_CURR_TEMP":  0xF, #Continuous bus voltage, shunt voltage and temperature
+    })
+    _adc_config_to_bin_mask = {"averaging_count": 0, "temperature":3, "current":6, "voltage":9, "mode": 12}
+
+    #SPI interface constants
+    spi_mode: ClassVar[int] = 1
+    min_spi_max_speed: ClassVar[float] = 1e7
+    max_spi_max_speed: ClassVar[float] = 0e1
+    spi_bit_order: ClassVar[str] = 'msb'
+    spi_word_bit_count: ClassVar[int] = 8
+
+    #ADC constants
+    reference_voltage: ClassVar[float] = 3.3
+    _max_queue_size = 1000
+    _voltage_conversion_factor: ClassVar[float] = 195.3125e-6 #195.3125 uV/LSB (2's complement)
+    _current_conversion_factor: ClassVar[float] = 0 #To be computed at init (known as 'CURRENT_LSB' in datasheet)
+    _temperature_conversion_factor: ClassVar[float] = 7.8125e-3 #7.8125 mC/LSB (2's complement)
+    _max_expected_current = 80 #Absolute maximum current (transient) needed to be measured
+
+    def __post_init__(self, nbr_avg_voltage:int=10, nbr_avg_current:int=10, nbr_avg_temperature:int=10) -> None:
```

Comment:
If you want to modify these, then you can add them as InitVar. But this change does not seem necessary for now.